From 588f818f9077dd0c6e058895cbe5efe8cefd7bc0 Mon Sep 17 00:00:00 2001
From: Martin Ritter <martin.ritter@lmu.de>
Date: Tue, 18 Sep 2018 17:50:01 +0200
Subject: [PATCH 1/5] [infra] Fix boost_python detection

If a user has a custom boost installation the boost python suffix might
be different than the platform default. However the option
`--with-boost-python-suffix` is only honoured on Darwin.

This makes the option override autodetection
---
 configure.ac | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/configure.ac b/configure.ac
index bfad5f24..6a9c1b8f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -259,10 +259,11 @@ AC_ARG_ENABLE([python],
 	[],
 	[])
 AM_CONDITIONAL([EOS_ENABLE_PYTHON], [test "x$enable_python" == "xyes"])
+has_boost_python_suffix=no
 AC_ARG_WITH([boost-python-suffix],
-	[AS_HELP_STRING([--with-boost-python-suffix], [suffix used to specify the libboost-python shared object])]
-	[],
-	[])
+	[AS_HELP_STRING([--with-boost-python-suffix], [suffix used to specify the libboost-python shared object])],
+	[has_boost_python_suffix=yes],
+	[with_boost_python_suffix=""])
 dnl }}}
 dnl {{{ python checks (you can change the required python version below)
 if test "x$enable_python" == xyes ; then
@@ -271,12 +272,14 @@ if test "x$enable_python" == xyes ; then
 	PYTHON_LIBS="-lpython$PYTHON_VERSION"
 	PYTHON_LDFLAGS="`python$PYTHON_VERSION-config --ldflags`"
 	PYTHON_CXXFLAGS="`python$PYTHON_VERSION-config --includes`"
-	os_id="`. /etc/os-release ; echo $ID`"
-	if test "x$os_id" == xubuntu -o "x$os_id" == xdebian ; then
-		BOOST_PYTHON_SUFFIX="-py`echo $PYTHON_VERSION | sed -e 's/\.//'`"
-	elif test "x$os_id" == xcentos ; then
-		BOOST_PYTHON_SUFFIX=""
-	elif test "x`uname`" == xDarwin ; then
+	if test "x$has_boost_python_suffix" == "xno"; then
+		os_id="`. /etc/os-release ; echo $ID`"
+		if test "x$os_id" == xubuntu -o "x$os_id" == xdebian ; then
+			BOOST_PYTHON_SUFFIX="-py`echo $PYTHON_VERSION | sed -e 's/\.//'`"
+		elif test "x$os_id" == xcentos ; then
+			BOOST_PYTHON_SUFFIX=""
+		fi
+	else
 		BOOST_PYTHON_SUFFIX=$with_boost_python_suffix
 	fi
 	AC_SUBST([PYTHON_LIBS])

From df93ca063d926384658e201739cc8f8f66175d96 Mon Sep 17 00:00:00 2001
From: Martin Ritter <martin.ritter@lmu.de>
Date: Tue, 18 Sep 2018 17:56:31 +0200
Subject: [PATCH 2/5] [infra] Use the actual python libs returned by
 `python-config --libs`

Otherwise eos might try to link against `libpython3.6` even though only
a library with the abi flag (`libpython3.6m`) is present
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 6a9c1b8f..4b017403 100644
--- a/configure.ac
+++ b/configure.ac
@@ -269,7 +269,7 @@ dnl {{{ python checks (you can change the required python version below)
 if test "x$enable_python" == xyes ; then
 	AM_PATH_PYTHON([3.5])
 	PY_PREFIX=`$PYTHON -c 'import sys ; print(sys.prefix)'`
-	PYTHON_LIBS="-lpython$PYTHON_VERSION"
+	PYTHON_LIBS="`python$PYTHON_VERSION-config --libs`"
 	PYTHON_LDFLAGS="`python$PYTHON_VERSION-config --ldflags`"
 	PYTHON_CXXFLAGS="`python$PYTHON_VERSION-config --includes`"
 	if test "x$has_boost_python_suffix" == "xno"; then

From 5cf170a9b55ae8e83fc688a1842e41250b8ea1b9 Mon Sep 17 00:00:00 2001
From: Martin Ritter <martin.ritter@lmu.de>
Date: Tue, 18 Sep 2018 18:42:18 +0200
Subject: [PATCH 3/5] [infra] Allow GSL in non-standard location

Use GSL_CXXFLAGS and GSL_LDFLAGS where necessary
---
 configure.ac                  | 2 ++
 eos/Makefile.am               | 2 ++
 eos/b-decays/Makefile.am      | 1 +
 eos/form-factors/Makefile.am  | 2 ++
 eos/optimize/Makefile.am      | 3 ++-
 eos/rare-b-decays/Makefile.am | 1 +
 eos/statistics/Makefile.am    | 6 +++++-
 eos/utils/Makefile.am         | 8 +++++---
 python/Makefile.am            | 2 +-
 src/clients/Makefile.am       | 9 +++++----
 10 files changed, 26 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4b017403..44bc164e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -203,6 +203,8 @@ AC_CHECK_LIB(gsl, gsl_sf_zeta,
 CXXFLAGS=$_CXXFLAGS
 LDFLAGS=$_LDFLAGS
 LIBS="$LIBS $_LIBS"
+AC_SUBST([GSL_LDFLAGS])
+AC_SUBST([GSL_CXXFLAGS])
 dnl }}}
 
 dnl {{{ use Minuit2 path if supplied via --with-minuit2=
diff --git a/eos/Makefile.am b/eos/Makefile.am
index cd58a2ec..a25c4fc8 100644
--- a/eos/Makefile.am
+++ b/eos/Makefile.am
@@ -13,6 +13,7 @@ libeos_la_SOURCES = \
 	decays.hh \
 	observable.cc observable.hh \
 	signal-pdf.cc signal-pdf.hh
+libeos_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 libeos_la_LIBADD = \
 	$(top_builddir)/eos/utils/libeosutils.la \
 	$(top_builddir)/eos/statistics/libeosstatistics.la \
@@ -48,6 +49,7 @@ check_PROGRAMS = \
 	references_TEST
 
 constraint_TEST_SOURCES = constraint_TEST.cc
+constraint_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 
 references_TEST_SOURCES = references_TEST.in
 
diff --git a/eos/b-decays/Makefile.am b/eos/b-decays/Makefile.am
index 09bb3b98..95620e0d 100644
--- a/eos/b-decays/Makefile.am
+++ b/eos/b-decays/Makefile.am
@@ -18,6 +18,7 @@ libeosbdecays_la_SOURCES = \
 	lambdab-to-lambdac2625-l-nu.cc lambdab-to-lambdac2625-l-nu.hh \
 	inclusive-b-to-u.cc inclusive-b-to-u.hh \
 	properties.cc properties.hh
+libeosbdecays_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) 
 libeosbdecays_la_LIBADD = \
 	$(top_builddir)/eos/utils/libeosutils.la \
 	$(top_builddir)/eos/form-factors/libeosformfactors.la \
diff --git a/eos/form-factors/Makefile.am b/eos/form-factors/Makefile.am
index 61eb49bb..a60c5525 100644
--- a/eos/form-factors/Makefile.am
+++ b/eos/form-factors/Makefile.am
@@ -17,6 +17,7 @@ libeosformfactors_la_SOURCES = \
 	mesonic.cc mesonic.hh mesonic-impl.hh \
 	pi-lcdas.cc pi-lcdas.hh \
 	zero-recoil-sum-rule.cc zero-recoil-sum-rule.hh
+libeosformfactors_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 libeosformfactors_la_LIBADD = \
 	$(top_builddir)/eos/utils/libeosutils.la \
 	-lgslcblas \
@@ -58,6 +59,7 @@ analytic_b_to_kstar_TEST_SOURCES = analytic-b-to-kstar_TEST.cc
 analytic_b_to_pi_TEST_SOURCES = analytic-b-to-pi_TEST.cc
 
 analytic_b_to_pi_pi_TEST_SOURCES = analytic-b-to-pi-pi_TEST.cc
+analytic_b_to_pi_pi_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 
 baryonic_TEST_SOURCES = baryonic_TEST.cc
 
diff --git a/eos/optimize/Makefile.am b/eos/optimize/Makefile.am
index 6c947217..e53ccb21 100644
--- a/eos/optimize/Makefile.am
+++ b/eos/optimize/Makefile.am
@@ -4,7 +4,7 @@ lib_LTLIBRARIES = libeosoptimize.la
 libeosoptimize_la_SOURCES = \
 	optimizer.cc optimizer.hh \
 	optimizer-gsl.cc optimizer-gsl.hh
-libeosoptimize_la_CXXFLAGS = $(AM_CXXFLAGS)
+libeosoptimize_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 libeosoptimize_la_LIBADD =
 
 include_eos_optimizedir = $(includedir)/eos/optimize
@@ -23,3 +23,4 @@ LDADD = \
 check_PROGRAMS = $(TESTS)
 
 optimizer_gsl_TEST_SOURCES = optimizer-gsl_TEST.cc
+optimizer_gsl_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
diff --git a/eos/rare-b-decays/Makefile.am b/eos/rare-b-decays/Makefile.am
index 99046965..8f3f4e9b 100644
--- a/eos/rare-b-decays/Makefile.am
+++ b/eos/rare-b-decays/Makefile.am
@@ -19,6 +19,7 @@ libeosrarebdecays_la_SOURCES = \
 	long-distance.cc long-distance.hh \
 	lambda-b-to-lambda-dilepton.cc lambda-b-to-lambda-dilepton.hh \
 	qcdf_integrals.cc qcdf_integrals.hh
+libeosrarebdecays_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) 
 libeosrarebdecays_la_LIBADD = \
 	$(top_builddir)/eos/utils/libeosutils.la \
 	$(top_builddir)/eos/form-factors/libeosformfactors.la \
diff --git a/eos/statistics/Makefile.am b/eos/statistics/Makefile.am
index 0522dc38..af5757b5 100644
--- a/eos/statistics/Makefile.am
+++ b/eos/statistics/Makefile.am
@@ -102,18 +102,22 @@ chi_squared_TEST_SOURCES = chi-squared_TEST.cc
 density_wrapper_TEST_SOURCES = density-wrapper_TEST.cc density-wrapper_TEST.hh
 
 hierarchical_clustering_TEST_SOURCES = hierarchical-clustering_TEST.cc
+hierarchical_clustering_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 
 histogram_TEST_SOURCES = histogram_TEST.cc
 
 log_likelihood_TEST_SOURCES = log-likelihood_TEST.cc
+log_likelihood_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 
 log_posterior_TEST_SOURCES = log-posterior_TEST.cc log-posterior_TEST.hh
-log_posterior_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(MINUIT2_CXXFLAGS)
+log_posterior_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) $(MINUIT2_CXXFLAGS)
 log_posterior_TEST_LDFLAGS = $(MINUIT2_LDFLAGS) -lMinuit2
 
 log_prior_TEST_SOURCES = log-prior_TEST.cc
+log_prior_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 
 markov_chain_TEST_SOURCES = markov-chain_TEST.cc
+markov_chain_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 
 markov_chain_sampler_TEST_SOURCES = markov-chain-sampler_TEST.cc density-wrapper_TEST.cc
 markov_chain_sampler_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS)
diff --git a/eos/utils/Makefile.am b/eos/utils/Makefile.am
index 50e2b2c9..0e68e3e1 100644
--- a/eos/utils/Makefile.am
+++ b/eos/utils/Makefile.am
@@ -76,8 +76,8 @@ libeosutils_la_LIBADD = \
 	-lMinuit2 \
 	-lpthread \
 	-lyaml-cpp
-libeosutils_la_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS) $(MINUIT2_CXXFLAGS)
-libeosutils_la_LDFLAGS = $(AM_CXXFLAGS) $(HDF5_LDFLAGS) $(MINUIT2_LDFLAGS)
+libeosutils_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) $(HDF5_CXXFLAGS) $(MINUIT2_CXXFLAGS)
+libeosutils_la_LDFLAGS = $(AM_CXXFLAGS) $(GSL_LDFLAGS) $(HDF5_LDFLAGS) $(MINUIT2_LDFLAGS)
 
 EXTRA_DIST = \
 	polylog_TEST_dilog.bin \
@@ -196,12 +196,13 @@ hdf5_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS)
 hdf5_TEST_LDFLAGS = $(AM_CXXFLAGS) $(HDF5_LDFLAGS)
 
 equation_solver_TEST_SOURCES = equation_solver_TEST.cc
-equation_solver_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(MINUIT2_CXXFLAGS)
+equation_solver_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(MINUIT2_CXXFLAGS) $(GSL_CXXFLAGS)
 equation_solver_TEST_LDFLAGS = $(MINUIT2_LDFLAGS)
 
 indirect_iterator_TEST_SOURCES = indirect-iterator_TEST.cc
 
 integrate_TEST_SOURCES = integrate_TEST.cc
+integrate_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 
 join_TEST_SOURCES = join_TEST.cc
 
@@ -248,6 +249,7 @@ top_loops_TEST_SOURCES = top-loops_TEST.cc
 verify_TEST_SOURCES = verify_TEST.cc
 
 wilson_coefficients_TEST_SOURCES = wilson_coefficients_TEST.cc
+wilson_coefficients_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 
 wilson_polynomial_TEST_SOURCES = wilson-polynomial_TEST.cc
 
diff --git a/python/Makefile.am b/python/Makefile.am
index ee2118f0..282aca52 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -38,7 +38,7 @@ eosplot_SCRIPTS = \
 _eosdir = $(pythondir)
 _eos_LTLIBRARIES = _eos.la
 _eos_la_SOURCES = _eos.cc
-_eos_la_CXXFLAGS = $(PYTHON_CXXFLAGS) @AM_CXXFLAGS@
+_eos_la_CXXFLAGS = $(PYTHON_CXXFLAGS) @AM_CXXFLAGS@ @GSL_CXXFLAGS@
 _eos_la_LDFLAGS = -module -avoid-version -export-symbols-regex PyInit__eos
 _eos_la_LIBADD = $(top_builddir)/eos/libeos.la $(top_builddir)/eos/utils/libeosutils.la -lboost_python$(BOOST_PYTHON_SUFFIX)
 
diff --git a/src/clients/Makefile.am b/src/clients/Makefile.am
index 21587172..ab1b9c72 100644
--- a/src/clients/Makefile.am
+++ b/src/clients/Makefile.am
@@ -58,7 +58,7 @@ endif
 eos_evaluate_SOURCES = eos-evaluate.cc
 
 eos_find_mode_SOURCES = eos-find-mode.cc
-eos_find_mode_CXXFLAGS = $(AM_CXXFLAGS) $(MINUIT2_CXXFLAGS)
+eos_find_mode_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) $(MINUIT2_CXXFLAGS)
 eos_find_mode_LDADD = $(LDADD) $(MINUIT2_LDFLAGS)
 
 eos_list_constraints_SOURCES = eos-list-constraints.cc
@@ -72,17 +72,18 @@ eos_list_signal_pdfs_SOURCES = eos-list-signal-pdfs.cc
 eos_print_polynomial_SOURCES = eos-print-polynomial.cc
 
 eos_propagate_uncertainty_SOURCES = eos-propagate-uncertainty.cc
-eos_propagate_uncertainty_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS)
+eos_propagate_uncertainty_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS) $(GSL_CXXFLAGS)
 eos_propagate_uncertainty_LDADD = $(LDADD) $(HDF5_LDFLAGS)
 
 eos_sample_mcmc_SOURCES = eos-sample-mcmc.cc
+eos_sample_mcmc_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 
 eos_sample_events_mcmc_SOURCES = eos-sample-events-mcmc.cc
-eos_sample_events_mcmc_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS)
+eos_sample_events_mcmc_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS) $(GSL_CXXFLAGS)
 eos_sample_events_mcmc_LDADD = $(LDADD) $(HDF5_LDFLAGS)
 
 eos_scan_mc_SOURCES = eos-scan-mc.cc
-eos_scan_mc_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS) $(MINUIT2_CXXFLAGS)
+eos_scan_mc_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS) $(GSL_CXXFLAGS) $(MINUIT2_CXXFLAGS)
 eos_scan_mc_LDADD = $(LDADD) $(HDF5_LDFLAGS) $(MINUIT2_LDFLAGS)
 
 integrated_SOURCES = integrated.cc

From 0158d86d07b4aae512ac6ad6c21061da3269d688 Mon Sep 17 00:00:00 2001
From: Danny van Dyk <danny.van.dyk@gmail.com>
Date: Sat, 24 Mar 2018 21:06:07 +0100
Subject: [PATCH 4/5] [infra] Fix building in environment with non-standard
 yaml-cpp location

Github: resolves #61
---
 configure.ac                  | 28 ++++++++++++++++++++++++++++
 eos/Makefile.am               |  3 +++
 eos/form-factors/Makefile.am  |  3 +--
 eos/rare-b-decays/Makefile.am |  3 +--
 eos/statistics/Makefile.am    |  6 +++---
 eos/utils/Makefile.am         |  5 ++---
 src/clients/Makefile.am       |  8 ++++----
 7 files changed, 42 insertions(+), 14 deletions(-)

diff --git a/configure.ac b/configure.ac
index 44bc164e..dbde3524 100644
--- a/configure.ac
+++ b/configure.ac
@@ -100,12 +100,40 @@ AC_CHECK_LIB([boost_system], main,
 	)
 dnl }}}
 dnl {{{ yaml-cpp
+PKG_CHECK_MODULES([YAMLCPP],
+	[yaml-cpp],
+	[
+		YAMLCPP_CXXFLAGS="$YAMLCPP_CFLAGS"
+		YAMLCPP_LDFLAGS="$YAMLCPP_LIBS"
+		AC_MSG_NOTICE([using pkg-config for yaml-cpp location])
+	],
+	[
+		YAMLCPP_CXXFLAGS=""
+		YAMLCPP_LDFLAGS="-lyaml-cpp"
+		AC_MSG_NOTICE([falling back to default for yaml-cpp location])
+	])
+
+AC_SUBST([YAMLCPP_CXXFLAGS])
+AC_SUBST([YAMLCPP_LDFLAGS])
+
+_CXXFLAGS=$CXXFLAGS
+_LDFLAGS=$LDFLAGS
+_LIBS=$LIBS
+
+CXXFLAGS="$YAMLCPP_CXXFLAGS $CXXFLAGS"
+LDFLAGS="$YAMLCPP_LDFLAGS $LDFLAGS"
+LIBS=""
+
 AC_CHECK_LIB([yaml-cpp], main)
 AC_CHECK_HEADERS([yaml-cpp/yaml.h],
         [eos_found_yaml_cpp_headers=yes; break;])
 
 AS_IF([test "x$eos_found_yaml_cpp_headers" != "xyes"],
         [AC_MSG_ERROR([you do seem to be lacking 'libyaml-cpp-dev'.])])
+
+CXXFLAGS=$_CXXFLAGS
+LDFLAGS=$_LDFLAGS
+LIBS=$_LIBS
 dnl }}}
 dnl }}}
 
diff --git a/eos/Makefile.am b/eos/Makefile.am
index a25c4fc8..63cd9abf 100644
--- a/eos/Makefile.am
+++ b/eos/Makefile.am
@@ -20,6 +20,8 @@ libeos_la_LIBADD = \
 	$(top_builddir)/eos/form-factors/libeosformfactors.la \
 	$(top_builddir)/eos/b-decays/libeosbdecays.la \
 	$(top_builddir)/eos/rare-b-decays/libeosrarebdecays.la
+libeos_la_CXXFLAGS = $(AM_CXXFLAGS) $(YAMLCPP_CXXFLAGS)
+libeos_la_LDFLAGS = $(AM_CXXFLAGS) $(YAMLCPP_LDFLAGS)
 
 include_eosdir = $(includedir)/eos
 include_eos_HEADERS = \
@@ -50,6 +52,7 @@ check_PROGRAMS = \
 
 constraint_TEST_SOURCES = constraint_TEST.cc
 constraint_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
+constraint_TEST_LDADD = $(LDADD) -lyaml-cpp
 
 references_TEST_SOURCES = references_TEST.in
 
diff --git a/eos/form-factors/Makefile.am b/eos/form-factors/Makefile.am
index a60c5525..579a0d41 100644
--- a/eos/form-factors/Makefile.am
+++ b/eos/form-factors/Makefile.am
@@ -21,8 +21,7 @@ libeosformfactors_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 libeosformfactors_la_LIBADD = \
 	$(top_builddir)/eos/utils/libeosutils.la \
 	-lgslcblas \
-	-lgsl \
-	-lyaml-cpp
+	-lgsl
 
 include_eos_rarebdecaysdir = $(includedir)/eos/form-factors
 include_eos_rarebdecays_HEADERS = \
diff --git a/eos/rare-b-decays/Makefile.am b/eos/rare-b-decays/Makefile.am
index 8f3f4e9b..a5d39abe 100644
--- a/eos/rare-b-decays/Makefile.am
+++ b/eos/rare-b-decays/Makefile.am
@@ -24,8 +24,7 @@ libeosrarebdecays_la_LIBADD = \
 	$(top_builddir)/eos/utils/libeosutils.la \
 	$(top_builddir)/eos/form-factors/libeosformfactors.la \
 	-lboost_filesystem -lboost_system \
-	-lgsl -lgslcblas -lm \
-	-lyaml-cpp
+	-lgsl -lgslcblas -lm
 
 include_eos_rarebdecaysdir = $(includedir)/eos/rare-b-decays
 include_eos_rarebdecays_HEADERS = \
diff --git a/eos/statistics/Makefile.am b/eos/statistics/Makefile.am
index af5757b5..f88ca004 100644
--- a/eos/statistics/Makefile.am
+++ b/eos/statistics/Makefile.am
@@ -35,9 +35,9 @@ libeosstatistics_la_SOURCES = \
 	simple-parameters.cc simple-parameters.hh \
 	test-statistic.cc test-statistic.hh test-statistic-impl.hh \
 	welford.cc welford.hh
-libeosstatistics_la_LIBADD = -lpthread -lgsl -lgslcblas -lm -lMinuit2
-libeosstatistics_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) $(HDF5_CXXFLAGS) $(MINUIT2_CXXFLAGS)
-libeosstatistics_la_LDFLAGS = $(AM_CXXFLAGS) $(GSL_LDFLAGS) $(MINUIT2_LDFLAGS)
+libeosstatistics_la_LIBADD = -lpthread -lgsl -lgslcblas -lm -lMinuit2 -lyaml-cpp
+libeosstatistics_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) $(HDF5_CXXFLAGS) $(MINUIT2_CXXFLAGS) $(YAMLCPP_CXXFLAGS)
+libeosstatistics_la_LDFLAGS = $(AM_CXXFLAGS) $(GSL_LDFLAGS) $(MINUIT2_LDFLAGS) $(YAMLCPP_LDFLAGS)
 
 if EOS_ENABLE_PMC
 libeosstatistics_la_SOURCES += \
diff --git a/eos/utils/Makefile.am b/eos/utils/Makefile.am
index 0e68e3e1..555edd42 100644
--- a/eos/utils/Makefile.am
+++ b/eos/utils/Makefile.am
@@ -76,8 +76,8 @@ libeosutils_la_LIBADD = \
 	-lMinuit2 \
 	-lpthread \
 	-lyaml-cpp
-libeosutils_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) $(HDF5_CXXFLAGS) $(MINUIT2_CXXFLAGS)
-libeosutils_la_LDFLAGS = $(AM_CXXFLAGS) $(GSL_LDFLAGS) $(HDF5_LDFLAGS) $(MINUIT2_LDFLAGS)
+libeosutils_la_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) $(HDF5_CXXFLAGS) $(MINUIT2_CXXFLAGS) $(YAMLCPP_CXXFLAGS)
+libeosutils_la_LDFLAGS = $(AM_CXXFLAGS) $(GSL_LDFLAGS) $(HDF5_LDFLAGS) $(MINUIT2_LDFLAGS) $(YAMLCPP_LDFLAGS)
 
 EXTRA_DIST = \
 	polylog_TEST_dilog.bin \
@@ -176,7 +176,6 @@ TESTS = \
 	wilson-polynomial_TEST \
 	wilson_scan_model_TEST
 LDADD = \
-	-lyaml-cpp \
 	$(top_builddir)/test/libeostest.a \
 	libeosutils.la \
 	$(top_builddir)/eos/libeos.la
diff --git a/src/clients/Makefile.am b/src/clients/Makefile.am
index ab1b9c72..635ff212 100644
--- a/src/clients/Makefile.am
+++ b/src/clients/Makefile.am
@@ -40,8 +40,7 @@ LDADD = \
 	$(top_builddir)/eos/libeos.la \
 	libcli.a \
 	-lboost_filesystem -lboost_system \
-	-lMinuit2 \
-	-lyaml-cpp
+	$(YAMLCPP_LDFLAGS)
 
 if EOS_ENABLE_PMC
 bin_PROGRAMS += \
@@ -58,10 +57,11 @@ endif
 eos_evaluate_SOURCES = eos-evaluate.cc
 
 eos_find_mode_SOURCES = eos-find-mode.cc
-eos_find_mode_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) $(MINUIT2_CXXFLAGS)
+eos_find_mode_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS) $(MINUIT2_CXXFLAGS) $(YAMLCPP_CXXFLAGS)
 eos_find_mode_LDADD = $(LDADD) $(MINUIT2_LDFLAGS)
 
 eos_list_constraints_SOURCES = eos-list-constraints.cc
+eos_list_constraints_CXXFLAGS = $(AM_CXXFLAGS) $(YAMLCPP_CXXFLAGS)
 
 eos_list_observables_SOURCES = eos-list-observables.cc
 
@@ -84,7 +84,7 @@ eos_sample_events_mcmc_LDADD = $(LDADD) $(HDF5_LDFLAGS)
 
 eos_scan_mc_SOURCES = eos-scan-mc.cc
 eos_scan_mc_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS) $(GSL_CXXFLAGS) $(MINUIT2_CXXFLAGS)
-eos_scan_mc_LDADD = $(LDADD) $(HDF5_LDFLAGS) $(MINUIT2_LDFLAGS)
+eos_scan_mc_LDADD = $(LDADD) $(HDF5_LDFLAGS) -lMinuit2 $(MINUIT2_LDFLAGS)
 
 integrated_SOURCES = integrated.cc
 

From 5e4deae4b857f629c699f0395c0e3b468338f7da Mon Sep 17 00:00:00 2001
From: Martin Ritter <martin.ritter@desy.de>
Date: Tue, 18 Sep 2018 19:05:45 +0200
Subject: [PATCH 5/5] [infra] Missing hdf5 dependency

Otherwise I get this

```
ld: eos_propagate_uncertainty-eos-propagate-uncertainty.o: undefined reference to symbol 'H5check_version'
ld: [...]/libhdf5.so.100.1.0: error adding symbols: DSO missing from command line
```
---
 eos/statistics/Makefile.am | 3 +++
 eos/utils/Makefile.am      | 1 +
 src/clients/Makefile.am    | 2 +-
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/eos/statistics/Makefile.am b/eos/statistics/Makefile.am
index f88ca004..0305810a 100644
--- a/eos/statistics/Makefile.am
+++ b/eos/statistics/Makefile.am
@@ -122,6 +122,7 @@ markov_chain_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
 markov_chain_sampler_TEST_SOURCES = markov-chain-sampler_TEST.cc density-wrapper_TEST.cc
 markov_chain_sampler_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS)
 markov_chain_sampler_TEST_LDFLAGS = $(AM_CXXFLAGS) $(HDF5_LDFLAGS)
+markov_chain_sampler_TEST_LDADD = $(LDADD) -lhdf5
 
 if EOS_ENABLE_PMC
 population_monte_carlo_sampler_TEST_SOURCES = population-monte-carlo-sampler_TEST.cc density-wrapper_TEST.cc
@@ -132,10 +133,12 @@ endif
 prior_sampler_TEST_SOURCES = prior-sampler_TEST.cc
 prior_sampler_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS)
 prior_sampler_TEST_LDFLAGS = $(AM_CXXFLAGS) $(HDF5_LDFLAGS)
+prior_sampler_TEST_LDADD = $(LDADD) -lhdf5
 
 proposal_functions_TEST_SOURCES = proposal-functions_TEST.cc
 proposal_functions_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS)
 proposal_functions_TEST_LDFLAGS = $(AM_CXXFLAGS) $(HDF5_LDFLAGS)
+proposal_functions_TEST_LDADD = $(LDADD) -lhdf5
 
 rvalue_TEST_SOURCES = rvalue_TEST.cc
 
diff --git a/eos/utils/Makefile.am b/eos/utils/Makefile.am
index 555edd42..72bb8ff7 100644
--- a/eos/utils/Makefile.am
+++ b/eos/utils/Makefile.am
@@ -193,6 +193,7 @@ derivative_TEST_SOURCES = derivative_TEST.cc
 hdf5_TEST_SOURCES = hdf5_TEST.cc
 hdf5_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS)
 hdf5_TEST_LDFLAGS = $(AM_CXXFLAGS) $(HDF5_LDFLAGS)
+hdf5_TEST_LDADD = $(LDADD) -lhdf5
 
 equation_solver_TEST_SOURCES = equation_solver_TEST.cc
 equation_solver_TEST_CXXFLAGS = $(AM_CXXFLAGS) $(MINUIT2_CXXFLAGS) $(GSL_CXXFLAGS)
diff --git a/src/clients/Makefile.am b/src/clients/Makefile.am
index 635ff212..b5104c21 100644
--- a/src/clients/Makefile.am
+++ b/src/clients/Makefile.am
@@ -73,7 +73,7 @@ eos_print_polynomial_SOURCES = eos-print-polynomial.cc
 
 eos_propagate_uncertainty_SOURCES = eos-propagate-uncertainty.cc
 eos_propagate_uncertainty_CXXFLAGS = $(AM_CXXFLAGS) $(HDF5_CXXFLAGS) $(GSL_CXXFLAGS)
-eos_propagate_uncertainty_LDADD = $(LDADD) $(HDF5_LDFLAGS)
+eos_propagate_uncertainty_LDADD = $(LDADD) -lhdf5 $(HDF5_LDFLAGS)
 
 eos_sample_mcmc_SOURCES = eos-sample-mcmc.cc
 eos_sample_mcmc_CXXFLAGS = $(AM_CXXFLAGS) $(GSL_CXXFLAGS)
